fighters/metaknight/src/meta_quick.rs

```diff
@@ -0,0 +1,205 @@
+// opff import
+use super::*;
+use globals::*;
+
+/// handles all of the meta quick logic
+pub unsafe fn run(fighter: &mut smash::lua2cpp::L2CFighterCommon) {
+    // update MeterModule
+    MeterModule::update(fighter.object(), false);
+    MeterModule::watch_damage(fighter.object(), true);
+    MeterModule::set_damage_gain_mul(fighter.object(), 6.0);
+    
+    println!("Meter Module: {}", MeterModule::meter(fighter.object()));
+    println!("Gimmick Timer: {}", VarModule::get_int(fighter.object(), vars::common::GIMMICK_TIMER));
+    
+
+
+    // if we have full meter, make meta quick available
+    if MeterModule::level(fighter.object()) >= 10 {
+        // if you press taunt while not in shield or dead, start meta quick
+        if !fighter.is_status_one_of(&[*FIGHTER_STATUS_KIND_GUARD, *FIGHTER_STATUS_KIND_GUARD_ON, *FIGHTER_STATUS_KIND_DEAD])
+            && fighter.is_cat_flag(Cat2::AppealAll) {
+
+            MeterModule::drain(fighter.object(), 10);
+            
+            // 8 seconds of quick per 50 damage
+            start_meta_quick(fighter, 8 * 60);
+        } else {
+            show_quick_ready_flash(fighter);
+        }
+    } else {
+        ColorBlendModule::cancel_main_color(fighter.boma(), 0);
+    }
+
+    // during meta quick, if you deal damage, it should extend meta quick
+    if is_meta_quick(fighter) && MeterModule::level(fighter.object()) >= 1 {
+        MeterModule::drain(fighter.object(), 1);
+        // an additional 0.5 seconds of quick per 5 damage dealt
+        VarModule::add_int(fighter.object(), vars::common::GIMMICK_TIMER, 30);
+    }
+    
+    check_apply_speeds(fighter);
+    update_meta_quick_timer(fighter);
+
+    // handle the main meta quick logic
+    if is_meta_quick(fighter) {
+        show_quick_active_effect(fighter);
+        check_reset(fighter);
+        // set the increased jump speed max multiplier for momentum transfer
+        VarModule::set_float(fighter.object(), vars::common::JUMP_SPEED_MAX_MUL, 1.5);
+    } else {
+        kill_quick_effect(fighter);
+    
+        // set the regular jump speed max multiplier for momentum transfer
+        VarModule::set_float(fighter.object(), vars::common::JUMP_SPEED_MAX_MUL, 0.75);
+
+        // if you are initial dash, slow them down slightly
+        if fighter.is_status(*FIGHTER_STATUS_KIND_DASH) {
+            let motion_vec = Vector3f {
+                x: -0.25 * PostureModule::lr(fighter.boma()) * (1.0 - (MotionModule::frame(fighter.boma()) / MotionModule::end_frame(fighter.boma()))),
+                y: 0.0, 
+                z: 0.0
+            };
+            KineticModule::add_speed_outside(fighter.boma(), *KINETIC_OUTSIDE_ENERGY_TYPE_WIND_NO_ADDITION, &motion_vec);
```

Comment:
pls don't use wind anymore lol, use sv_kinetic_energy funcs (add_speed in this case)

Reply:
i used this for a reason but i can no longer remember what it is so I guess ill just look at it again